memgpt/memory.py

```diff
@@ -124,7 +126,10 @@ def core_memory_replace(self, name: str, old_content: str, new_content: str) ->
         Returns:
             Optional[str]: None is always returned as this function does not produce a response.
         """
-        self.memory[name].value = self.memory[name].value.replace(old_content, new_content)
+        if old_content == "":
+            self.memory[name].value = new_content
```

Comment:
@Vandracon is this logic correct?

Doesn't this mean if the agent calls `core_memory_replace("", "hello")`, it'll wipe all of core memory and replace it with `"hello"`?

Reply:
Yeah, I wasn't sure of the intended design since I was working to get it not making a huge string. After thinking about it some more, throwing a validation error makes more sense and making the AI less capable of emptying their core memory is a better approach. I just pushed an update that reflects this and can confirm it returns an error to the AI if they try to send old_content = "".

Reply:
Also, I think it needs a sqlite migration to repair the function in the database for existing agents. Was less comfortable making one for sqlite / postgres / idk what else. But even without that, this fixes two things.